src/app/components/OptimoPromos/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useContext } from 'react';
+import { node, string, shape } from 'prop-types';
+import { ServiceContext } from '#contexts/ServiceContext';
+import path from 'ramda/src/path';
+
+import Timestamp from '#components/Promo/timestamp';
+import LiveLabel from '#legacy/psammead-live-label/src';
+import useViewTracker from '#hooks/useViewTracker';
+
+import Title from './Title';
+import Link from './Link';
+import MediaIndicator from './MediaIndicator';
+import Content from './Content';
+import { PromoWrapper, ContentWrapper, BorderWrapper } from './Wrappers.styles';
+import Image from './Image';
+
+const PromoContext = React.createContext({});
+
+const Promo = ({ children, toLink, a11yId, mediaType, eventTrackingData }) => {
+  const { service } = useContext(ServiceContext);
+
+  const eventTrackingDataSend = path(['block'], eventTrackingData);
+
+  const viewRef = useViewTracker(eventTrackingDataSend);
+
+  return (
+    <BorderWrapper>
+      <PromoWrapper ref={viewRef}>
+        <PromoContext.Provider
+          value={{ service, toLink, a11yId, eventTrackingData, mediaType }}
+        >
+          {children}
+        </PromoContext.Provider>
+      </PromoWrapper>
+    </BorderWrapper>
+  );
+};
+
+const withPromoContext = Component => props =>
+  (
+    <PromoContext.Consumer>
+      {context => <Component {...context} {...props} />}
```

Comment:
props spreading here

Reply:
what made you go with providing context like this rather than getting from the context using `useContext`?

Reply:
Maybe my execution is not the best, but coming from this [conversation](https://github.com/bbc/simorgh/pull/10165#pullrequestreview-1023599949) with @andrewscfc